xbmc/cores/VideoPlayer/VideoRenderers/OverlayRenderer.cpp

```diff
@@ -580,25 +602,21 @@ void CRenderer::LoadSettings()
   m_subtitleHorizontalAlign = static_cast<SubtitleHorizontalAlign>(
       CServiceBroker::GetSettingsComponent()->GetSettings()->GetInt(
           CSettings::SETTING_SUBTITLES_CAPTIONSALIGN));
+  ResetSubtitlePosition();
 }
 
-int CRenderer::GetSubtitleVerticalMargin()
+double CRenderer::GetSubtitleVerticalMarginPerc()
 {
-  int subAlign = CServiceBroker::GetSettingsComponent()->GetSettings()->GetInt(
-      CSettings::SETTING_SUBTITLES_ALIGN);
+  // We return the vertical margin as percentage
+  // to fit the current screen resolution
+  const std::shared_ptr<CSettings> settings{CServiceBroker::GetSettingsComponent()->GetSettings()};
+  int subAlign{settings->GetInt(CSettings::SETTING_SUBTITLES_ALIGN)};
 
   // If the user has set the alignment type to keep the subtitle text
-  // inside the black bars, we exclude custom vertical margin
-  // and we force our fixed margin to try avoid go off the black bars
+  // inside the black bars, we override user vertical margin
+  // to try avoid go off the black bars
   if (subAlign == SUBTITLE_ALIGN_BOTTOM_OUTSIDE || subAlign == SUBTITLE_ALIGN_TOP_OUTSIDE)
     return KODI::SUBTITLES::MARGIN_VERTICAL_BLACKBARS;
 
-  // Try get the vertical margin customized by user
-  int overrideMerginVertical =
-      CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_videoSubtitleVerticalMargin;
-
-  if (overrideMerginVertical >= 0)
-    return overrideMerginVertical;
-
-  return KODI::SUBTITLES::MARGIN_VERTICAL;
+  return settings->GetNumber(CSettings::SETTING_SUBTITLES_MARGINVERTICAL);
```

Comment:
Would be nice to wrap this as a getter in SubtitlesSettings to avoid spreading raw setting access everywhere (better encapsulation)

Reply:
add method try look